### PR TITLE
configs: Add defconfigs for esp-sparrow-kit

### PR DESCRIPTION
**Pull request by andreeaThePenguin**
## Summary

Added defconfigs to xtensa/esp32/esp32-sparrow-kit for wifi and mqttc enablement.

## Impact

N/A

## Testing

Tested configs + applications with esp32-sparrow-kit board

### CODE REVIEW

**Comment by nuttxpr** [comment]
[**\[Experimental Bot, please feedback here\]**](https://github.com/search?q=repo%3Aapache%2Fnuttx+13552&type=issues)



No, this PR does not fully meet the NuttX requirements.  While it provides a summary of the change, it lacks crucial details.

Here's what's missing:

* **Summary:**  Needs more detail. *Why* are these defconfigs necessary?  What problem do they solve or what new functionality do they enable?  What part of the code is changed (e.g., Kconfig files)? *How* do the changes work (what options are enabled in the defconfigs)?
* **Impact:**  Claiming "N/A" is almost never accurate. At a minimum, this adds new build configurations.  Does this affect any existing functionality? Does it change the size of the build?  Think carefully about all the impact categories and justify why they are N/A, or describe the actual impact.
* **Testing:**  This is insufficient.  "Tested configs + applications" is vague.  What applications?  What tests were run?  Provide *specific* commands used and the *actual output* (log snippets) before and after the change to demonstrate the fix or new functionality.  Include details about the build host and target as requested.  Just saying "it works" is not enough.

The PR needs significant revision to meet the NuttX requirements.  Providing the missing details will greatly improve the chances of the PR being accepted.

File: boards/xtensa/esp32/esp32-sparrow-kit/configs/mqttc/defconfig [hunk 1]
```diff
@@ -0,0 +1,147 @@
+#
+# This file is autogenerated: PLEASE DO NOT EDIT IT.
+#
+# You can use "make menuconfig" to make any modifications to the installed .config file.
+# You can then do "make savedefconfig" to generate a new defconfig file that includes your
+# modifications.
```
Comment (anchao) [inline boards/xtensa/esp32/esp32-sparrow-kit/configs/mqttc/defconfig:6]: Please refresh defconfig by script:
`./nuttx/tools/refresh.sh --silent boards/xtensa/esp32/esp32-sparrow-kit/configs/mqttc/`

**Comment by acassis** [comment]
Hi @andreeaThePenguin thank you for this example. I noticed we are missing a Documentation for this board, it is not listed at https://nuttx.apache.org/docs/latest/platforms/xtensa/esp32/index.html#supported-boards

Please submit later a Documentation and include information about this mqttc profile you added, this way people will know how to test it.

File: boards/xtensa/esp32/esp32-sparrow-kit/configs/mqttc/defconfig (hunk 1 above)
Comment (andreeaThePenguin) [reply boards/xtensa/esp32/esp32-sparrow-kit/configs/mqttc/defconfig:6]: Done & commited

**Comment by andreeaThePenguin** [comment]
Hi @acassis, I will include a short description of the board in the Documentation later, similar to the boards in the link. I will also add information on the submitted profiles.

As a reference until I add them, they can be tested similarly to the ones from ESP32 DevKitC: https://nuttx.apache.org/docs/latest/platforms/xtensa/esp32/boards/esp32-devkitc/index.html

**Review by anchao — APPROVED** [review]

**Review by xiaoxiang781216 — APPROVED** [review]

**Comment by acassis** [comment]
> Hi @acassis, I will include a short description of the board in the Documentation later, similar to the boards in the link. I will also add information on the submitted profiles.
> 
> As a reference until I add them, they can be tested similarly to the ones from ESP32 DevKitC: https://nuttx.apache.org/docs/latest/platforms/xtensa/esp32/boards/esp32-devkitc/index.html

You can use that board as reference, just modifying the name, pictures and including your board profile description

**Pull request merged**